tests/test_ble_lifecycle_controller.py

```
@@ -0,0 +1,689 @@
+        controller = BLELifecycleController(mock_iface)
+        controller._disconnect = mock_disconnect  # type: ignore[assignment]
+
+        result = controller._handle_disconnect("source")
+
+        assert result is False
+
+    def test_on_ble_disconnect_delegates_to_disconnect_coordinator(self) -> None:
+        """_on_ble_disconnect should delegate to the _disconnect coordinator."""
+        calls: list[object] = []
+        mock_disconnect = SimpleNamespace(
+            on_ble_disconnect=lambda client: calls.append(client)
```

Ruff (0.15.6)

[warning] Lambda may be unnecessary; consider inlining inner function

Inline function call

(PLW0108)